cpp/memilio/utils/uncertain_value.h

```diff
@@ -227,6 +227,15 @@ class UncertainValue
     std::unique_ptr<ParameterDistribution> m_dist;
 };
 
+/**
+ * @brief Format UncertainValues using their value for logging with spdlog.
+ */
+template <class FP>
+double format_as(const UncertainValue<FP>& uv)
```

Comment:
don't we then also need a format_as for an ad type with UV and derivative?

Reply:
Good catch, that "double" should be a "FP" (or better "const FP&"). fmt is able to call the AD format_as if an AD type is returned here.